src/gromacs/nblib/simulationstate.h

```diff
@@ -102,6 +102,32 @@ class SimulationState
     std::vector<gmx::RVec> velocities_;
 };
 
+class SimulationState
+{
+public:
+    SimulationState(const std::vector<gmx::RVec>& coord,
+                    Box                           box,
+                    Topology                      topology,
+                    const std::vector<gmx::RVec>& velocities = {}) :
+        simState_impl_ptr_(std::make_shared<SimulationState_impl>(coord, box, topology, velocities))
+    {}
+
+    //! Returns topology of the current state
+    const Topology& topology() const { return simState_impl_ptr_->topology(); }
+
+    //! Returns the box
+    const Box& box() { return simState_impl_ptr_->box(); }
+
+    //! Returns a vector of particle coordinates
+    std::vector<gmx::RVec>& coordinates() { return simState_impl_ptr_->coordinates(); }
+
+    //! Returns a vector of particle velocities
+    std::vector<gmx::RVec>& velocities() { return simState_impl_ptr_->velocities(); }
+
+private:
+    std::shared_ptr<SimulationState_impl> simState_impl_ptr_;
```

Comment:
I think this should be `simulationStatePtr_`, since we use CamelCase and the `ulation` addition to `sim` doesn't hurt.

Reply:
Done.